src/app/state/types.ts

```diff
@@ -49,3 +49,37 @@ export type CurrentView =
   | { type: 'story-options'; subview: 'story-metadata' | 'story-wide-code' | 'asset-upload' }
   | { type: 'scene'; id: string; subview: 'scene-options' | '3d-view' }
   | { type: 'preview'; previous?: CurrentView };
+
+export type Visibility = 'public' | 'private';
+
+export interface Creator {
+  email: string;
+  id: string;
+  name: string;
+}
+
+export interface BaseItem {
```

Comment:
Are the supported fields checked at the backend? (something like `extra="forbid"` on a pydantic model in the API)

Don't want the users to be able to store arbitrary data by doing direct API calls.

Reply:
Let's use a different extension than .js. It's msgpack object.

10MB for the json and 50mb doesn't make much sense, since that session file will also contain the assets. Is there a way to enforce file limit on the session files though? Eg you can have at most 10?

To validate the content: 
- valid msgpack file for session 
- json file for mvsj
- zip file with index.mvsj for mvsx?